From a8812558f0af4c5785e4b51b9ca36ed6f228d672 Mon Sep 17 00:00:00 2001
From: Potnuri Bharat Teja <bharat@chelsio.com>
Date: Fri, 6 Oct 2017 17:56:14 +0530
Subject: [PATCH 38/48] iw_cxgb4: clean up c4iw_reject_cr()

commit bce2841f5a0dd9caf327fb1bc0051ea09a9e7dfb
Author: Hariprasad S <hariprasad@chelsio.com>
Date:   Fri Jun 10 01:05:13 2016 +0530

	RDMA/iw_cxgb4: clean up c4iw_reject_cr()

	Get rid of unneeded code, and refactor things a bit.

	For MPA version 0 we abort the connection.  For > 0, we attempt to send
	an MPA_START/REJECT Reply, and then disconnect gracefully.  If the send
	of the MPA message fails, then we abort the connection.  We can ignore
	c4iw_ep_disconnect() errors here because it will clean up the endpoint
	if there are failures.

	Signed-off-by: Steve Wise <swise@opengridcomputing.com>
	Signed-off-by: Hariprasad Shenai <hariprasad@chelsio.com>
	Signed-off-by: Doug Ledford <dledford@redhat.com>
---
 drivers/infiniband/hw/cxgb4/cm.c | 18 +++++++-----------
 1 file changed, 7 insertions(+), 11 deletions(-)

diff --git a/drivers/infiniband/hw/cxgb4/cm.c b/drivers/infiniband/hw/cxgb4/cm.c
index 149d6caa3c64..ca56609dd842 100644
--- a/drivers/infiniband/hw/cxgb4/cm.c
+++ b/drivers/infiniband/hw/cxgb4/cm.c
@@ -3040,8 +3040,7 @@ out:
 
 int c4iw_reject_cr(struct iw_cm_id *cm_id, const void *pdata, u8 pdata_len)
 {
-	int err = 0;
-	int disconnect = 0;
+	int abort;
 	struct c4iw_ep *ep = to_ep(cm_id);
 
 	pr_debug("ep %p tid %u\n", ep, ep->hwtid);
@@ -3054,16 +3053,13 @@ int c4iw_reject_cr(struct iw_cm_id *cm_id, const void *pdata, u8 pdata_len)
 	}
 	set_bit(ULP_REJECT, &ep->com.history);
 	if (mpa_rev == 0)
-		disconnect = 2;
-	else {
-		err = send_mpa_reject(ep, pdata, pdata_len);
-		disconnect = 1;
-	}
+		abort = 1;
+	else
+		abort = send_mpa_reject(ep, pdata, pdata_len);
 	mutex_unlock(&ep->com.mutex);
-	if (disconnect) {
-		stop_ep_timer(ep);
-		err = c4iw_ep_disconnect(ep, disconnect == 2, GFP_KERNEL);
-	}
+
+	stop_ep_timer(ep);
+	c4iw_ep_disconnect(ep, abort != 0, GFP_KERNEL);
 	c4iw_put_ep(&ep->com);
 	return 0;
 }
-- 
2.5.3

